### PR TITLE
nix/regenerate.sh: Fix .stack-to-nix.cache bug

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

Found during ADP-1053.

### Overview

- Bumps Haskell.nix to fix that annoying error in nix-tools:
      ```
      stack-to-nix: .stack-to-nix.cache: openFile: resource busy (file is locked)
      ```
       (input-output-hk/haskell.nix#1184, input-output-hk/nix-tools#98)

- Adds @jbgi's cabal-wrapper (input-output-hk/iohk-nix#489) to the nix-shell so that we don't need a separate `cabal-nix.project` any more.

### CODE REVIEW

**Review by sevanspowell — APPROVED** [review]
Took it for a spin, looks good to me!

File: shell.nix [hunk 1]
```diff
@@ -60,6 +62,11 @@ let
 
     meta.platforms = lib.platforms.unix;
   };
+
+  helpCommand = with pkgs.buildPackages; writeShellScriptBin "help" ''
```
Comment (sevanspowell) [inline shell.nix]: Is this just for fun?

Comment (rvl) [reply shell.nix]: Placeholder command -- removed now.

**Pull request merged**